### PR TITLE
Itay/ex5

### PR DESCRIPTION
**Pull request by itayneu**

### CODE REVIEW

File: src/dist/main.js [hunk 1]
```diff
@@ -24,26 +24,60 @@ class Main {
...
+
   renderItems = async () => {
     const list = document.getElementById("list");
     list.innerHTML = "";
 
-    const todoList = await this.itemClient.getItems();
-    const items = todoList.map((item) => {
-      return item.todo;
-    });
+    const itemsList = await this.itemClient.getItems();
```
Comment (SergeiSafrigin) [inline src/dist/main.js:38]: avoid fetching all the items per item update/add

File: src/dist/main.js [hunk 2]
```diff
@@ -24,26 +24,60 @@ class Main {
     await this.renderItems();
   };
 
+  updateItem = async (item) => {
+    item.status = !item.status;
```
Comment (SergeiSafrigin) [inline src/dist/main.js:28]: add a separate functions for updating status (which will call this method)

File: src/dist/main.js [hunk 3]
```diff
@@ -24,26 +24,60 @@ class Main {
...
 
-    items.forEach((item) => {
+    itemsList.forEach((item) => {
+      const itemName = item.itemName;
       const listItem = document.createElement("li");
       listItem.classList.add("list-item");
-      listItem.innerHTML = item;
 
-      const listItemDeleteButton = this._createDeleteButton(item);
+      const listItemCheckbox = this._createCheckbox(item);
```
Comment (SergeiSafrigin) [inline src/dist/main.js:45]: extract the createItemElement logic to a different function

File: src/dist/main.js [hunk 4]
```diff
@@ -24,26 +24,60 @@ class Main {
...
+
+  _createLabel = (item) => {
+    const label = document.createElement("label");
+    label.classList.add("list-item-label");
+
+    label.innerText =
+      item.status === true
+        ? `${item.itemName} (Done: ${new Date(
+            item.updatedAt
+          ).toLocaleString()})`
```
Comment (SergeiSafrigin) [inline src/dist/main.js:75]: consider using a different technique for showing the done date (tooltip? title dom attribute?)

File: src/server/db/config/config.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "host": "127.0.0.1",
+    "dialect": "mysql"
+  },
+  "test": {
+    "username": "root",
+    "password": null,
+    "database": "database_test",
+    "host": "127.0.0.1",
+    "dialect": "mysql"
+  },
```
Comment (SergeiSafrigin) [inline src/server/db/config/config.json:15]: like!

File: src/server/db/migrations/20220618101015-create-items.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('Items', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      itemName: {
```
Comment (SergeiSafrigin) [inline src/server/db/migrations/20220618101015-create-items.js:11]: consider changing to name

File: src/server/services/item_manager.js [hunk 7]
```diff
@@ -1,102 +1,88 @@
...
 }
 
-async function writeTodoFile(content) {
-  try {
-    await fs.writeFile(todoFile, JSON.stringify(content));
-  } catch (error) {
-    console.error(`Failed to write to file ${error.message}`);
-  }
+async function updateTodoItem(input) {
+  console.log("input", input);
```
Comment (SergeiSafrigin) [inline src/server/services/item_manager.js:72]: remove

File: src/server/services/item_manager.js [hunk 8]
```diff
@@ -1,102 +1,88 @@
...
+  await Items.update(
```
Comment (SergeiSafrigin) [inline src/server/services/item_manager.js:73]: separate the business logic ("flow-manager") from the data access layer logic

File: src/server/services/item_manager.js [hunk 9]
```diff
@@ -1,102 +1,88 @@
...
 
 async function addTodoItem(input) {
   const inputString = input.trim();
-  let dataArray = await getTodoItems();
 
-  if (!dataArray) dataArray = [];
   // number
   if (/^[0-9]+$/.test(inputString)) {
-    handlePokemonItem(inputString, dataArray);
+    handlePokemonItem(inputString);
```
Comment (SergeiSafrigin) [inline src/server/services/item_manager.js:12]: consider changing to a more descriptive name + more different from addPokemonItem

File: src/server/services/item_manager.js [hunk 10]
```diff
@@ -1,102 +1,88 @@
...
   }
   // comma separated list of IDs
   else if (/^[0-9, ]+$/.test(inputString)) {
-    handlePokemonItems(inputString, dataArray);
+    handlePokemonItems(inputString);
   }
   // normal todo
   else {
-    handleNormalItem(inputString, dataArray);
+    handleNormalItem(inputString);
```
Comment (SergeiSafrigin) [inline src/server/services/item_manager.js:20]: textItem/ baseItem

**Review by SergeiSafrigin — CHANGES_REQUESTED** [review]
Good job! Overall it looks good!
Keep:
- A good improvement in separation of concerns 💪

Improve:
- A bit more emphasis on the UI/UX will skyrocket your app 😁
- Make sure the app works and running out of the box (if a running MySQL via docker is required, kindly add the instructions to the readme file)

File: src/server/routes/todoRouter.js [hunk 11]
```diff
@@ -1,11 +1,12 @@
 const express = require("express");
-const { createItem, getItems, deleteItem } = require("./api");
+const { createItem, getItems, deleteItem, updateItem } = require("./api");
 const { validateSchema, todoSchema } = require("../middleware/validation");
 
 const todoRouter = express.Router();
 
 todoRouter.get("/items", getItems);
```
Comment (SergeiSafrigin) [inline src/server/routes/todoRouter.js:7]: server.js is not in the changelist, but don't forget to fix the static path that serves the dist folder

File: src/dist/main.js (hunk 2 above)
Comment (itayneu) [reply src/dist/main.js:28]: Done

**Pull request closed without merging**